kapitan/inputs/jinja2.py

```diff
@@ -34,13 +31,18 @@ def compile_file(self, file_path, compile_path, ext_vars, **kwargs):
         reveal = kwargs.get("reveal", False)
         target_name = kwargs.get("target_name", None)
 
+        input_params = self.input_params
+        # set compile_path allowing jsonnet to have context on where files
+        # are being compiled on the current kapitan run
+        # we only do this if user didn't pass its own value
+        input_params.setdefault("compile_path", compile_path)
+
         # set ext_vars and inventory for jinja2 context
         context = ext_vars.copy()
         context["inventory"] = inventory(self.search_paths, target_name)
         context["inventory_global"] = inventory(self.search_paths, None)
-        context["input_params"] = self.input_params
-        # reset between each compile if jinja2 component is used multiple times
```

Comment:
I'm confused. Why is this not desired?

Reply:
As explained offline, resetting the input_params causes the bug. Example:

```yaml
    compile:
      - output_path: test-1/      <---   We only set input_params at creation of each compile obj kapitan/targets.py:L477
        input_type: jinja2
        input_params:
          name: test1
          namespace: ns1
        input_paths:
          - templates/pod.yml     <---   reset happens after here
          - templates/service.yml <---   here input_params is empty (changed name to give a possible use case)
 ```